config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -188,6 +190,14 @@ final ExpressionBasedFilterInvocationSecurityMetadataSource createMetadataSource
 			if (trustResolver != null) {
 				defaultHandler.setTrustResolver(trustResolver);
 			}
+			ApplicationContext context = http.getSharedObject(ApplicationContext.class);
+			if(context != null) {
+				String[] roleHiearchyBeanNames = context.getBeanNamesForType(RoleHierarchy.class);
+				if(roleHiearchyBeanNames.length == 1) {
```

Comment:
It makes sense to do it only for the scenario where there is one bean. Multiple beans means the order is not deterministic. This would cause very strange errors if in dev they get once bean and then deploy to production and get a different Bean.

If users need multiple beans they can always explicitly wire up the instance they wish to use.